homeassistant/components/enasolar/sensor.py

```diff
@@ -0,0 +1,250 @@
+    remove = None
+    interval = min_interval
+
+    async def interval_listener(now=None):  # pylint: disable=unused-argument
+        """Handle elapsed interval with backoff."""
+        nonlocal interval, remove
+
+        try:
+            if await action():
+                interval = min_interval
+            else:
+                interval = min(interval * 2, SCAN_MAX_INTERVAL)
+        finally:
+            remove = async_call_later(hass, interval, interval_listener)
+
+    hass.async_create_task(interval_listener())
+
+    def remove_listener():
+        """Remove interval listener."""
+        remove()  # pylint: disable=not-callable
+
+    return remove_listener
+
+
+class EnaSolarSensor(SensorEntity):
+    """Representation of a EnaSolar sensor."""
+
+    def __init__(self, pyenasolar_sensor, inverter_name=None, serial_no=None):
+        """Initialize the EnaSolar sensor."""
+        self.sensor = pyenasolar_sensor
+        if inverter_name:
+            self._attr_name = f"{inverter_name}_{self.sensor.name}"
+        else:
+            self._attr_name = f"enasolar_{self.sensor.name}"
+        self.serial_no = serial_no
+        self._native_value = self.sensor.value
+
+        if pyenasolar_sensor.is_meter:
+            self._attr_state_class = STATE_CLASS_MEASUREMENT
+        else:
+            self._attr_state_class = STATE_CLASS_TOTAL_INCREASING
+        self._attr_native_unit_of_measurement = ENASOLAR_UNIT_MAPPINGS[self.sensor.unit]
+        self._attr_should_poll = False
+
+    @property
+    def name(self) -> str | None:
+        """Return the name of the sensor."""
+        if hasattr(self, "_attr_name"):
+            return self._attr_name
+        return None
+
+    @property
+    def native_value(self):
+        """Return the current sensor value."""
+        return self._native_value
+
+    @property
+    def device_class(self):
+        """Return the device class the sensor belongs to."""
+        if self._attr_native_unit_of_measurement == POWER_KILO_WATT:
+            return DEVICE_CLASS_POWER
+        if self._attr_native_unit_of_measurement == ENERGY_KILO_WATT_HOUR:
+            return DEVICE_CLASS_ENERGY
+        if self._attr_unit_of_measurement in (TEMP_CELSIUS, TEMP_FAHRENHEIT):
+            return DEVICE_CLASS_TEMPERATURE
+        return None
+
+    @callback
+    def async_update_values(self, unknown_state=False):
+        """Update this sensor."""
+        update = False
+
+        if self.sensor.value != self._native_value:
+            update = True
+            self._native_value = self.sensor.value
+
+        if unknown_state and self._native_value is not None:
+            update = True
+            self._native_value = None
+
+        if update:
+            self.async_write_ha_state()
+
+    @property
+    def unique_id(self):
+        """Return a unique identifier for this sensor."""
+        return f"{self.serial_no}_{self.sensor.name}"
```

Comment:
What is name?

Reply:
I assumed that this was the unique id of the sensor, not the device so in order to create something unique I used the devices serial number and the sensor's name e.g. energy_today_total => 1234567890_energy_today_total.